### PR TITLE
GKE deployment

### PR DESCRIPTION
**Pull request by nholbrook**
## Description
The PR includes the necessary configuration files to deploy my application to GKE. The kubernetes application is structured as follows:
- 1 Primary Pod  (and Service) containing the Food Finder and Supplier docker containers
- 3 Vendor Pods (and Services), with each Pod containing a distinct Vendor (although this number could be expanded to contain as many Vendors as necessary)
  - This structure (of having a separate pod for each vendor) was chosen to replicate a real world scenario where Vendors are accessed remotely from the Food Finder service.

Additionally, I added functionality for Vendors to dynamically register themselves with the Food Finder service when they boot up, who then retains a list of all available vendors and their IPs for subsequent calls.

To create a new Vendor, you need to create it's own Deployment and Service with GKE. Additionally, to configure it's inventory you need to create and configure a ConfigMap (see example for required formatting).

## Local Requirements
[Docker](https://www.docker.com) - v19.03.8

## Running Locally
1. From the root directory, run `docker build -t food-finder-client -f client/Dockerfile .`
2. Then run `docker run --rm -e "SERVICE_URL=104.196.112.54:50051" -e "FOOD_NAME=pepper" food-finder-client`, where the SERVICE_URL is the address of the Food Finder Service and the FOOD_NAME is the requested food. For testing use one of [milk, eggs, bread, pepper, butter]

## TODO
*For a complete list of pending tasks refer to the [dev kanban board](https://github.com/nholbrook/opentelemetry-starter-project/projects/1)*

**Remaining project goals include:**
- Integrate Open Census tracing / logging

### CODE REVIEW

File: proto/foodfinder.proto [hunk 1]
```diff
@@ -77,4 +78,7 @@ message Item {
   string name = 1;
   float price = 2;
   uint32 quantity = 3;
-}
\ No newline at end of file
+}
+
+// Empty message for one way grpc calls 
+message Empty {}
```
Comment (ziqizh) [inline proto/foodfinder.proto:84]: Just FYI: There is a pre-defined message type `google.protobuf.Empty`. To use it, just `import "google/protobuf/empty.proto";` that the beginning of the proto file, and `using google::protobuf::Empty;` in your cpp files.

File: client/Dockerfile [hunk 2]
```diff
@@ -2,11 +2,14 @@ FROM grpc/cxx:1.12.0
...
+ENV SERVICE_URL="localhost:50051"
+
 COPY client/client.cc ./client/
 COPY proto/foodfinder.proto ./proto/
 COPY Makefile ./
 
 RUN make food-finder-client && make clean
 
-CMD ["./food-finder-client"]
+CMD ["sh", "-c", "./food-finder-client $FOOD_NAME $SERVICE_URL"]
```
Comment (ziqizh) [inline client/Dockerfile:14]: If I want to query some food other than milk, do I need to build a new image? My interactive solution is to write a interactive command line interface and run locally.

File: proto/foodfinder.proto (hunk 1 above)
Comment (nholbrook) [reply proto/foodfinder.proto]: I saw that before but couldn't get the import working so I just created one of my own. I took a look at how you did it and finally got it working - thanks,

File: client/Dockerfile (hunk 2 above)
Comment (nholbrook) [reply client/Dockerfile:14]: Nope, you can just pass in an env variable when running it from the command line, e.g. `docker run --rm -e "SERVICE_URL=104.196.112.54:50051" -e "FOOD_NAME=pepper" food-finder-client`.

File: ff-server.yaml [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      containers:
+        - name: ff-vendor1-service
+          image: gcr.io/cal-intern-project/nholbrook-ff-vendor-service:latest
+          ports:
+            - containerPort: 50060
+          env:
+            - name: VENDOR_NAME
+              value: Meijer
+            - name: VENDOR_ADDRESS
+              value: 34.75.101.194:50060
```
Comment (jsandler18) [inline ff-server.yaml:91]: where does this address come from?

Comment (nholbrook) [reply ff-server.yaml]: I was hardcoding them, but I've updated it to fetch the current IP using kubectl.

**Pull request merged**